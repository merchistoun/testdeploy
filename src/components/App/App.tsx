import { useTranslate } from "../../localization/useTranslate";
import "./App.css";

const App = () => {
  const { translate } = useTranslate();
  return <h1>{translate("test")}</h1>;
};

export default App;
