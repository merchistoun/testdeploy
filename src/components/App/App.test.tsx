import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import App from "./App";

// filepath: src/components/App/App.test.tsx

// Mock the useTranslate hook
vi.mock("../../localization/useTranslate", () => ({
  useTranslate: () => ({
    translate: vi.fn((key: string) => `Translated: ${key}`),
  }),
}));

describe("App Component", () => {
  it("renders the translated text", () => {
    render(<App />);
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toMatchSnapshot("Translated: test");
  });
});
