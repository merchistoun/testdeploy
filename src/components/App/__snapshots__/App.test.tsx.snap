// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > renders the translated text > Translated: test 1`] = `
<h1>
  Translated: test
</h1>
`;
